core/src/main/scala/org/http4s/parser/ForwardedHeader.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s.parser
+
+import java.nio.charset.{Charset, StandardCharsets}
+import java.util.Locale
+
+import org.http4s.headers.Forwarded
+import org.http4s.internal.bug
+import org.http4s.internal.parboiled2._
+import org.http4s.internal.parboiled2.support._
+import org.http4s.{ParseResult, Uri => UriModel}
+
+private[parser] trait ForwardedHeader {
+  def FORWARDED(value: String): ParseResult[Forwarded] = new Parser(value).parse
+
+  private class Parser(value: String)
+      extends Http4sHeaderParser[Forwarded](value)
+      with Rfc3986Parser {
+
+    override def charset: Charset = StandardCharsets.ISO_8859_1
+
+    override def entry: Rule1[Forwarded] =
+      rule {
+        oneOrMore(ELEMENT).separatedBy(ListSep) ~ EOL ~> { (elems: Seq[Forwarded.Element]) =>
+          Forwarded(elems.head, elems.tail: _*)
+        }
+      }
+
+    // TODO: remove this override once the original `ListSet` gets fixed.
+    //       See also #3657.
+    override def ListSep: Rule0 = rule(OptWS ~ ',' ~ OptWS)
```

Comment:
If by chance we get #3743 merged before this one, it may make sense to remove this override.

Reply:
Well, it did. :smile: